builtin/logical/database/path_creds_create.go

```diff
@@ -249,10 +249,19 @@ func (b *databaseBackend) pathStaticCredsRead() framework.OperationFunc {
 		respData := map[string]interface{}{
 			"username":            role.StaticAccount.Username,
 			"ttl":                 role.StaticAccount.CredentialTTL().Seconds(),
-			"rotation_period":     role.StaticAccount.RotationPeriod.Seconds(),
 			"last_vault_rotation": role.StaticAccount.LastVaultRotation,
 		}
 
+		if role.StaticAccount.UsesRotationPeriod() {
+			respData["rotation_period"] = role.StaticAccount.RotationPeriod.Seconds()
+		} else if role.StaticAccount.UsesRotationSchedule() {
+			respData["rotation_schedule"] = role.StaticAccount.RotationSchedule
+		}
+
+		if role.StaticAccount.RotationWindow.Seconds() != 0 {
```

Comment:
Should we put this inside the `else if` block above? It is a very defensive move since this should not be set if we did our checks correctly on Create/Update. But it isn't relevant if we are using rotation period. What do you think?